src/literal.h

```diff
@@ -706,6 +707,40 @@ struct GCData {
   GCData(HeapType type, Literals values) : type(type), values(values) {}
 };
 
+// Wasm has nondeterministic rules for NaN propagation in some operations. For
```

Comment:
Maybe move the comment and declaration here, but the body can remain in the C++ I think - I don't believe this must be inlined for speed.

Reply:
(though it might be worth measuring)

Reply:
done!